hashintel/hash
# Implement draft state for entity editor links table

## 🌟 What is the purpose of this PR?
 This PR implements draft state for links table in entity editor. So the link entities will be created/archived when user hits `Save Changes` / `Create Entity` button. This will provide a better UX because it will works as same as properties table.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543021352032/1203711964051433/f) _(internal)_


## 🔍 What does this change?

- Add `useDraftLinkState` - creates draft link states
- Add `useApplyDraftLinkEntityChanges` - returns a function which applies the draft changes stored in the draft link states.
- Add `useMarkLinkEntityToArchive` - returns a function which marks the given link entity to archive. Created this to reuse the following logic: "checking if given link entity is a draft or a real entity" 

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Open entity editor for an entity type with links
1.  Confirm that links table works as expected while creating and editing entity (only applies the changes when user hits "save change" button, and discards the changes on "discard" button)

## 📹 Demo

Uploading Screen Recording 2023-01-16 at 15.33.16.mov…
